### PR TITLE
Screen autocomplete

### LINKED ISSUE #48: Support auto-complete in screen editor
This will require adding the ACE editor which is better since it supports line numbers

### PR DESCRIPTION
**Pull request by jmthomas**
closes #48

### CODE REVIEW

File: openc3-cmd-tlm-api/app/controllers/script_autocomplete_controller.rb [hunk 1]
```diff
@@ -45,17 +54,38 @@ def get_ace_autocomplete_data
...
+        {
+          :caption => keyword,
+          :snippet => snippet,
+          :meta => 'widget',
+        }
+      end
+    else
+      autocomplete_data = OpenC3::TargetModel.all(scope: scope).flat_map do |target_name, target_info|
+        OpenC3::TargetModel.packets(target_name, type: type.upcase.intern, scope: scope).flat_map do |packet|
+          packet_to_autocomplete_hashes(packet, target_info, type)
```
Comment (ryanmelt) [inline openc3-cmd-tlm-api/app/controllers/script_autocomplete_controller.rb:84]: This isn't going to scale well to lots of packets.

File: openc3-init/plugins/packages/openc3-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue [hunk 2]
```diff
@@ -169,6 +171,9 @@ export default {
         }
         this.updateScreens()
       })
+    Api.get('/openc3-api/autocomplete/keywords/screen').then((response) => {
+      this.keywords = response.data
```
Comment (ryanmelt) [inline openc3-init/plugins/packages/openc3-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue:175]: What about autocomplete for target,packet,item in valueWidgets?

**Review by ryanmelt — APPROVED** [review]

**Review by ryanmelt — CHANGES_REQUESTED** [review]
Lots of playwright failures still (and long run time)

File: openc3-cmd-tlm-api/app/controllers/script_autocomplete_controller.rb (hunk 1 above)
Comment (jmthomas) [reply openc3-cmd-tlm-api/app/controllers/script_autocomplete_controller.rb:84]: Worked ok for 1000 cmd packets generated by loadsim

File: openc3-init/plugins/packages/openc3-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue (hunk 2 above)
Comment (jmthomas) [reply openc3-init/plugins/packages/openc3-tool-tlmviewer/src/tools/TlmViewer/TlmViewer.vue:175]: Yeah I punted on that for now. Might be a way to incorporate the existing stuff in there.

**Pull request merged**